Chore add script for downloading render tools for development

kbeker opened the pull request:
Resolves #1127

Must be merged before #1128

dybi commented on download_render_tools_from_golem_repository.sh:
```diff
@@ -0,0 +1,47 @@
+
+for argument in "$@"
+do
+    if [[ ( $argument == "--help") ||  $argument == "-h" ]]; then
```
for me it is a bit unusual to see `if` and `switch` mixed like that

kbeker replied:
I did it exactly the same as we have it in `run_tests.sh`: https://github.com/golemfactory/concent/blob/master/run_tests.sh#L18

dybi commented on download_render_tools_from_golem_repository.sh:
```diff
@@ -0,0 +1,47 @@
+        -p=*|--full_path_to_store_files=*)
+        PATH_TO_STORE_FILES="${argument#*=}"
+        ;;
+        -r=*|--repository_tag=*)
```
shouldn't this be a separate `case` entry?

kbeker replied:
I did it exactly the same as we have it in `run_tests.sh`: https://github.com/golemfactory/concent/blob/master/run_tests.sh#L31

dybi commented on install-development-requirements.sh:
```diff
@@ -23,3 +23,7 @@ cd ..
 printf "\n"
+
+printf "=================== DOWNLOAD RENDER_TOOLS FROM GOLEM REPOSITORY ====================\n"
+./download_render_tools_from_golem_repo.sh
```
:heart:

cameel commented on download_render_tools_from_golem_repository.sh:
```diff
@@ -0,0 +1,47 @@
+ORGANIZATION_NAME="golemfactory"
+REPOSITORY_NAME="golem"
+REPOSITORY_TAG="develop"
+PATH_TO_STORE_FILES="/home/$USER/golem_render_tools"
```
The usual convention in Bash is to user lowercase names for your own variables local to the script. Only the system and configuration variables defined in the environment (like `PATH` or `PWD`) should be capitalized.

kbeker replied:
Changed to local variables to lowercase names

cameel commented on download_render_tools_from_golem_repository.sh:
```diff
@@ -0,0 +1,47 @@
+ORGANIZATION_NAME="golemfactory"
+REPOSITORY_NAME="golem"
+REPOSITORY_TAG="develop"
+PATH_TO_STORE_FILES="/home/$USER/golem_render_tools"
```
Why is this an absolute path? I think it would be better to assume that the script is being run from a working copy of Concent and put it in a location relative to that.

kbeker replied:
Changed to `concent_api/verifier/golem_render_tools`

cameel commented on download_render_tools_from_golem_repository.sh:
```diff
@@ -0,0 +1,47 @@
+REPOSITORY_NAME="golem"
+REPOSITORY_TAG="develop"
+PATH_TO_STORE_FILES="/home/$USER/golem_render_tools"
+FILES_TO_DOWNLOAD=("scenefileeditor.py" "templates/blendercrop.py.template")
```
You can break lines in bash lists like this:
``` bash
FILES_TO_DOWNLOAD=(
    "scenefileeditor.py"
    "templates/blendercrop.py.template"
)
```

kbeker replied:
Done

cameel commented on download_render_tools_from_golem_repository.sh:
```diff
@@ -0,0 +1,47 @@
+
+ORGANIZATION_NAME="golemfactory"
+REPOSITORY_NAME="golem"
+REPOSITORY_TAG="develop"
```
It does not have to be a tag so the name is a bit misleading. It can also be a commit or branch. We're usually referring to this as "version" in deployment.

kbeker replied:
Done, changed to `version`

cameel commented on download_render_tools_from_golem_repository.sh:
```diff
@@ -0,0 +1,47 @@
+
+ORGANIZATION_NAME="golemfactory"
+REPOSITORY_NAME="golem"
+REPOSITORY_TAG="develop"
```
Since `REPOSITORY_TAG` is just the default, I think you should have "default" in its name to make it clear.

kbeker replied:
Done

cameel commented on download_render_tools_from_golem_repository.sh:
```diff
@@ -0,0 +1,47 @@
+
+for argument in "$@"
+do
+    if [[ ( $argument == "--help") ||  $argument == "-h" ]]; then
```
The parenthesis is not necessary here.

kbeker replied:
Done

cameel commented on download_render_tools_from_golem_repository.sh:
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash -e
+
+
+display_usage(){
```
To be consistent with our conventions in deployment please use the `function` keyword when defining a function and leave a space before the `{`.

kbeker replied:
Done

cameel commented on download_render_tools_from_golem_repository.sh:
```diff
@@ -0,0 +1,47 @@
+        exit 0
+    fi
+    case $argument in
+        -p=*|--full_path_to_store_files=*)
```
Please use dashes in option names: `--full-path-to-store-files`.

kbeker replied:
Done

cameel commented on concent_api/verifier/utils.py:
```diff
@@ -36,7 +36,7 @@
 
 
-sys.path.append('/golem/entrypoints/scripts/render_tools')
+sys.path.append(os.path.join(settings.BLENDER_RENDER_TOOLS_DIR, '/golem/entrypoints/scripts/render_tools'))
```
I don't think utils is a good location for this. If I was looking for something like this I would not think to look here. Why not put it in some more obvious location, e.g. in the file with verifier's views or even somewhere in `concent_api/concent_api/`? Actually the `manage.py` file might be the best location though I'm not sure if you'll be able to import config there without any annoying side-effects.

kbeker replied:
Ok, moved to `manage.py`.

cameel commented on concent_api/concent_api/settings/base.py:
```diff
@@ -372,4 +372,4 @@
 
 # Define dir where render_tools from golem repository should be stored
-BLENDER_RENDER_TOOLS_DIR = ''
+# BLENDER_RENDER_TOOLS_DIR = ''
```
I think this should have a default in `base.py`. It should point at the directory where these files would if we were running in our Docker container.

kbeker replied:
Done

cameel commented on install-development-requirements.sh:
```diff
@@ -25,5 +25,5 @@ pip install -r requirements-development.txt
 
 printf "=================== DOWNLOAD RENDER_TOOLS FROM GOLEM REPOSITORY ====================\n"
-./download_render_tools_from_golem_repo.sh
+./download_render_tools_from_golem_repository.sh
```
Please use dashes in Bash script names: `download-render-tools-from-golem-repository.sh`.

kbeker replied:
Done

cameel commented on concent_api/verifier/utils.py:
```diff
@@ -36,7 +36,7 @@
 
 
-sys.path.append('/golem/entrypoints/scripts/render_tools')
+
```
Why is this change visible as removal (instead of just disappearing? Is this already in master?

kbeker replied:
This change is visible because it was added in previous PR. In this PR I am changing it to work with custom directory

cameel commented on download-render-tools-from-golem-repository.sh:
```diff
@@ -0,0 +1,53 @@
+    "templates/blendercrop.py.template"
+)
+
+DOWNLOAD_URL="https://raw.githubusercontent.com/${organization_name}/${repository_name}/${version}/apps/blender/resources/images/entrypoints/scripts/render_tools"
```
This should be lower case too.

kbeker replied:
Ups, I missed it. Done

cameel commented on .gitignore:
```diff
@@ -118,3 +118,6 @@ ENV/
 daily_thresholds/
+
+# Render tools downloaded from golem repository
+/concent_api/verifier/golem_render_tools
```
Please add a trailing slash so that it's clear it's a directory.

kbeker replied:
Done

kbeker commented:
Also, @cameel I changed where path is added. I decided that better place than `manage.py` and `wsgi.py` would be `__init__.pl` file in `verifier` directory. I have also added setting `CUSTOM_BLENDER_RENDER_TOOLS` which in production default is False.
